frontend/components/site/siteBuildLogs.jsx

```diff
@@ -13,26 +13,38 @@ export const REFRESH_INTERVAL = 15 * 1000;
 const BUILD_LOG_RETENTION_LIMIT = 180 * 24 * 60 * 60 * 1000; // 180 days in milliseconds
 
 function buildTooOld(build) {
-  return (new Date() - new Date(build.completedAt)) > BUILD_LOG_RETENTION_LIMIT;
+  return new Date() - new Date(build?.completedAt) > BUILD_LOG_RETENTION_LIMIT;
 }
 
 function getSiteBuildLogTable(buildDetails, logs, state) {
+  if (buildTooOld(buildDetails)) {
+    return (
+      <SiteBuildLogTable
+        buildLogs={[
+          'Builds more than 180 days old are deleted according to platform policy.',
+        ]}
+      />
+    );
+  }
+
   if (logs && logs?.length > 0) {
     return <SiteBuildLogTable buildLogs={logs} buildState={state} />;
   }
-  if (buildTooOld(buildDetails)) {
-    return <SiteBuildLogTable buildLogs={['Builds more than 180 days old are deleted according to platform policy.']} />;
-  }
-  return <SiteBuildLogTable buildLogs={['This build does not have any build logs.']} />;
+
+  return (
+    <SiteBuildLogTable
+      buildLogs={['This build does not have any build logs.']}
+    />
+  );
 }
 
 const SiteBuildLogs = () => {
   const { buildId: buildIdStr } = useParams();
   const buildId = parseInt(buildIdStr, 10);
-  const { logs, state } = useBuildLogs(buildId);
-  const { buildDetails, isLoading } = useBuildDetails(buildId);
+  const { buildDetails, isLoading: isLoadingBuildDetails } = useBuildDetails(buildId);
+  const { logs, state, isLoading: isLoadingBuildLogs } = useBuildLogs(buildId);
 
-  if (isLoading || !logs) {
+  if (isLoadingBuildDetails || isLoadingBuildLogs || !buildDetails) {
```

Comment:
What's the case where we are done loading build details but also don't have them?

Reply:
Waiting on `buildDetails` prevents a rendering flash since the `getSiteBuildLogTable` function depends on both `buildDetails` and  the `useBuildLogs` results to determine how the component will be displayed.

Reply:
Covered via voice, can be removed